test/unit/proc_macro/proc_macro_test.bzl

```diff
@@ -0,0 +1,100 @@
+"""Unittest to verify proc-macro targets"""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest")
+load("//rust:defs.bzl", "rust_proc_macro", "rust_test")
+load("//test/unit:common.bzl", "assert_action_mnemonic", "assert_list_contains_adjacent_elements", "assert_list_contains_adjacent_elements_not")
+
+def _proc_macro_test_targets(edition):
+    """Define a set of `rust_proc_macro` targets for testing
+
+    Args:
+        edition (str): The rust edition to use for the new targets
+    """
+    rust_proc_macro(
+        name = "proc_macro_lib_{}".format(edition),
+        srcs = [
+            "proc_macro_{}.rs".format(edition),
+        ],
+        edition = edition,
+    )
+
+    rust_test(
+        name = "proc_macro_lib_{}_unittest".format(edition),
+        crate = ":proc_macro_lib_{}".format(edition),
+        edition = edition,
+    )
+
+    rust_test(
```

Comment:
Please move the integration test to `//test/proc_macro`. Let's only keep unit tests in //test/unit/...

Reply:
Why aren't they all just considered tests and go in the root of test? It feels like we're gonna exponentially gain nearly identical rust files if we need to split everything like this.

Reply:
We should have order of magnitude more unit tests than integration tests. If they are duplicated, delete the integration test.

Reply:
I think the thing that bothers me is that targets and unit tests are declare together. I'd like to think of unit tests as assertions on existing targets but that I could organize it such that I have a bunch of targets in `test/*` and then potentially *only* define unit test targets (which wouldn't be `rust_*` targets) in `test/unit`. But maybe that doesn't make sense (I struggle greatly with starlark unittests and really dislike them 😅)

Reply:
Updated. But I think the coupling (decoupling) here will become unmaintainable if we continue with this pattern. I see no reason why there's a need for the `test/unit` directory. I would say each `test/*` directory defines integration tests as the base targets that are used for the relevant unit tests in the same package or a `unit` subpackage.

Reply:
Oh I'd like to go the other way. I'm bothered that the test setup and assertions are too far away from each other (and test setup is below assertions in the file, that is also bad). Ideally I'd like to write a test like:


```
def test_that_yolo_is_foo(ctx):
  ctx.package("//foo", """
    rust_library(name = "asdf", srcs = ["..."])
    rust_test(name="asdf_test", crate=":asdf")
  """
  argv = ctx.target("//foo:asdf_test").actions[0].argv
  assert_list_contains_adjacent_elements("--extern", "yolo")
```

And even more ideally this test would not execute any actions.